### ISSUE TITLE
nested-interactive: Improve failure description and fix message for nested checkboxes

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#3232]
The example below seems to work well with all the screen readers but it is in accessible using Dragon. When I command Dragon to click the checkbox, I see Dragon showing multiple options to select from despite only one checkbox being visible on the screen. 

Demo: https://codepen.io/iamrafan/pen/MWvjRVj?editors=1000

![image](https://user-images.githubusercontent.com/22068434/138280634-60d9a152-aeb2-4594-ada8-dc2a98e4fc6d.png)

### GIF showing the experience with Dragon: 
 ![nested-interactive issue using Dragon](https://user-images.githubusercontent.com/22068434/138281019-a2864467-1e08-4e70-a1f1-860666a903bc.gif)

Product: axe-core

Expectation: Show appropriate ```Failure description``` and ```Fix the following``` messages

Actual: The failure description only states the nested element is not available to screen readers

Motivation: This would be an issue for Speech recognition software users as they would not be sure which number to select to be able to click on the checkbox.

<pre><code>
axe-core version: 4.3.3
</code></pre

**Comment by padmavemulapati** [comment]
Verified with the latest dev branch code, 
for `nested-interactive` rule, failure description has changed, and now its describing as
`"Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies"`
![image](https://user-images.githubusercontent.com/47409579/147951188-4a3626d5-a95a-4ab2-aaf8-a5114a155639.png)
